src/ecosystem/corals/growth.jl

```diff
@@ -860,7 +860,7 @@ Calculates coral recruitment for each species/group and location.
 function recruitment_rate(larval_pool::AbstractArray{T,2}, A::AbstractArray{T};
     α::Union{T,Vector{T}}=2.5, β::Union{T,Vector{T}}=5000.0)::Matrix{T} where {T<:Float64}
 
-    sd = settler_density.(α, β, larval_pool) .* A
+    sd = settler_density.(α, β, larval_pool) .* A'
```

Comment:
Obviously this works but puzzled as to why the transform is necessary when it wasn't before?

I don't see a change in what is passed into `recruitment_rate` (in this PR at least)

Reply:
tl dr;
Before we were doing, inside scenario.jl 

```
loc_coral_cover = dropdims(sum(C_cover_t; dims=(1, 2)), dims=1)
```

and now this is being done inside the function `_loc_coral_cover` as :

```
dropdims(sum(C_cover_t; dims=(1, 2)), dims=(1, 2))
```

----- 

This is because `A`, that is called `leftover_space` inside the  function`settler_cover`, comes from `leftover_space_m²`, in `scenario.jl`: https://github.com/open-AIMS/ADRIA.jl/blob/d45cade0b6ca8f3a88edef47764f09d399289fe7/src/scenario.jl#L724

The argument that is being passed to the function `relative_leftover_space` is `loc_coral_cover`, which now is being computed inside the new function `_loc_coral_cover`. I implemented this function because we do this operation in different places so I extracted that to a function.

To standardize I had to drop dims 1 and 2, and not only 2.